daemon/redis.c

```diff
@@ -1582,7 +2058,66 @@ int redis_restore(struct callmaster *m, struct redis *r) {
 }
 
 
+static int json_update_crypto_params(JsonBuilder *builder, const char *pref, const str *callid,
+		unsigned int unique_id,
+		const char *key, const struct crypto_params *p)
+{
+	char tmp[2048]; ZERO(tmp);
+
+	if (!p->crypto_suite)
+		return -1;
+
+	sprintf(tmp,"%s-crypto_suite",key);
+	json_builder_set_member_name (builder, tmp);
+	ZERO(tmp);
+	json_builder_add_string_value (builder, p->crypto_suite->name);
+
+	sprintf(tmp,"%s-master_key",key);
+	json_builder_set_member_name (builder, tmp);
+	ZERO(tmp);
+	snprintf(tmp,sizeof(p->master_key), "%s",p->master_key);
+	json_builder_add_string_value (builder, tmp);
+	ZERO(tmp);
+
+	sprintf(tmp,"%s-master_salt",key);
+	json_builder_set_member_name (builder, tmp);
+	ZERO(tmp);
+	snprintf(tmp,sizeof(p->master_salt), "%s",p->master_salt);
+	json_builder_add_string_value (builder, tmp);
+	ZERO(tmp);
+
+	sprintf(tmp,"%s-unenc-srtp",key);
+	json_builder_set_member_name (builder, tmp);
+	ZERO(tmp);
+	sprintf(tmp,"%i",p->session_params.unencrypted_srtp);
```

Comment:
I suppose integer values are sprintf'd into strings and then stored as strings (instead of storing them as json integers) because the restore functions expect strings and then parse them into integers?